contracts/programs/bubblegum/src/lib.rs

```diff
@@ -567,14 +566,15 @@ pub mod bubblegum {
         }?;
         let merkle_slab = ctx.accounts.merkle_slab.to_account_info();
         emit!(voucher.leaf_schema.to_event());
-        insert_or_append_leaf(
+        replace_leaf(
```

Comment:
I think it makes sense for `cancel_redeem` to only attempt to replace given that the nonce (index) is known